### PR TITLE
FEAT: LINK GOOGLE CLOUD STORAGE DATABASE WITH MATERIAL MODEL

### PR DESCRIPTION
**Pull request by TomasCB24**
I have implemented the functionality to upload files of less than 5 MB (pdfs, presentations...) using the external storage provided by Google Cloud with free credits.

_**General Changes:**_

- Most names have been modified, changing "sample-microservice" to "learning-microservice" and other aspects in .env.dev, .env.prod, docker-compose, package-lock.json, package.json files.

- A number of dependencies and packages needed to interact with the files have been installed, namely "multer" for local file management, "uuid" for string generation to name these packages uniquely and "google-cloud/storage" to interact with google cloud. 
The commands used are the following (among them, to include them in the application dependencies)
 

> - npm install @google-cloud/storage multer to install both google cloud and multer

> - npm i --save-dev @types/multer

> - npm install --save-dev @types/uuid

- Included TODO comments for future enhancements.
- Added a .json in the "server" folder for interaction with Google Cloud.

**_Specific Changes in server/routes/materials.ts:_**

- Added imports of installed packages/libraries.

- At the start of materials.ts the storage variable has been set using the json key and the google cloud bucket.

- Set the limits that a file has to have 

- The post method has been modified to allow the upload of a specific file.

- Delete method has been modified to delete both in mongo database and in the bucket.

**_Checklist:_**
 

- [x]   Check that the "GoogleCloudKey.json" file is in the right place

- [x]   Check that the modified CRUD operations are working as expected.

- [x]   Check the .env.dev, .env.prod... files modified to suit our microservice, everything has been modified and there are no conflicts.

- [x]   Check if the weight limits of a material file comply with the standards.

### CODE REVIEW

File: server/routes/materials copy.ts
Comment (mrarjona) [inline server/routes/materials copy.ts:1]: This file should be deleted, it is duplicated.

File: server/routes/materials.ts [hunk 1]
```diff
@@ -33,75 +54,168 @@ router.get('/:id/users', async (req: Request, res: Response) => {
     return res.status(404).json({ error: 'Material not found' })
 })
 
-router.post('/', async (req: Request, res: Response) => {
-    const { title, description, price, file, type }: MaterialInputs = req.body
-
-    if (!title || !description || !price || !file || !type) {
+router.post('/', upload.single('file'), async (req: Request, res: Response) => {
```
Comment (mrarjona) [inline server/routes/materials.ts:57]: You must validate the inserted fields (title, description, price, type) before uploading the file. Right now if I try to upload a file with an incorrect field the material is not created but it is uploaded to the Google Cloud bucket.

File: server/routes/materials.ts [hunk 2]
```diff
@@ -33,75 +54,168 @@ router.get('/:id/users', async (req: Request, res: Response) => {
...
-        type,
-    })
-
-    const savedMaterial = await newMaterial.save()
-
-    res.status(201).json(savedMaterial)
-})
-
-router.put('/:id', async (req: Request, res: Response) => {
+router.put('/:id',  upload.single('file'), async (req: Request, res: Response) => {
```
Comment (mrarjona) [inline server/routes/materials.ts:114]: The same happens as in the POST operation, you must validate that the inserted fields (title, description, price, type) are correct before uploading the file. Right now if I try to change the file with an incorrect field the material is not updated in the database but the old file is deleted and the new one is uploaded in the Google Cloud bucket.

**Review by mrarjona — CHANGES_REQUESTED** [review]
I have required a couple of changes, associated with deleting a duplicate file and validating the form fields before making changes to the Google Cloud bucket.

File: server/routes/materials copy.ts
Comment (TomasCB24) [reply server/routes/materials copy.ts:1]: true

**Review by mrarjona — APPROVED** [review]
Thank you very much for solving everything I told you, now everything works correctly. Incredible work Tomás! 💪🏻

**Pull request merged**